testplan/web_ui/testing/src/AssertionPane/AssertionTypes/TextAttachment.js

```diff
@@ -0,0 +1,270 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {StyleSheet, css} from 'aphrodite';
+import axios from 'axios';
+import SyntaxHighlighter from 'react-syntax-highlighter';
+import { FadeLoader } from 'react-spinners';
+
+const DISPLAY_NUM = 20
+const SCROLLBAR_MIN_LIMIT = 60
+
+/**
+ * TextAttachment component:
+ *   * Gets text file using Axios
+ *   * Returns JSX depending on how long the text file is
+ */
+
+class TextAttachment extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+          loading: false,
+          error: undefined,
+          textContent: "",
+          originalText: undefined,
+          numberOfLines: undefined,
+          expandButtonPushed: false,
+          scrollbar: undefined,
+        };
+    }
+
+    /**
+    * Fetch the text attachment once the component has mounted.
+    * @public
+    */
+    componentDidMount() {
+        this.setState({loading: true});
+        this.getTextAttachment()
+    }
+
+    /*
+     * Prepare the contents of the collapsed .txt content, by displaying the
+     * last DISPLAY_NUM lines of the text file (or the entire text file if
+     * it has less lines than DISPLAY_NUM - short files).
+     *
+     * @param {array} text_array (each element of array is one line of text)
+     * @return {jsx} Content for the .txt file content when collapsed
+     * @private
+     */
+    getCollapsedText(text_array){
+        let last_lines;
+        let i;
+        let starting_line;
+        let num_of_lines = text_array.length;
+
+        if(num_of_lines < DISPLAY_NUM){
+            i = 0;
+            starting_line = 1;
+            last_lines = "";
+        }
+        else{
+            i = num_of_lines - DISPLAY_NUM;
+            starting_line = num_of_lines - DISPLAY_NUM;
+            last_lines = "...\n";
+        }
+
+        for(i; i < num_of_lines; i++){
+            last_lines = last_lines + text_array[i] + '\n';
+        }
+
+        let return_jsx = (
+            <SyntaxHighlighter showLineNumbers
+                startingLineNumber={starting_line}>
+              {last_lines}
+            </SyntaxHighlighter>
+        )
+
+        return return_jsx
+    }
+
+    /**
+    * Fetch the text attachment.
+    *   * Handle UID errors.
+    *   * Make a GET request for the .txt attachment file.
+    *   * Prepare the display content
+    * @public
+    */
+    getTextAttachment() {
+    const paths = window.location.pathname.split('/');
+
+    // Check if dev mode and display test_string
+    if ((paths.length >= 2) && (paths[1] === '_dev')) {
+        const TEST_TEXT = "test1\ntest2\ntest3\ntest4\ntest5\ntest6\ntest7\n";
+        let text = TEST_TEXT;
+        let lines = text.split('\n');
+        let length = lines.length;
+
+        if(lines.pop() === ""){
+            text = text + "<newline>"
+            lines[length-1] = "<newline>"
+        }
+
+        this.setState({
```

Comment:
Can we merge these two setState calls into one? I think that each setState call can potentially trigger a separate re-render, so it is more efficient to just call it once.